src/renderer/services/wallet/hdwallet.ts

```diff
@@ -0,0 +1,23 @@
+import * as O from 'fp-ts/lib/Option'
+
+import { observableState } from '../../helpers/stateHelper'
+import { INITIAL_HDWALLET_STATE } from './const'
+import { HDWalletService, HDWalletState } from './types'
+
+const { get$: getHDWalletState$, set: setHDWalletState } = observableState<HDWalletState>(INITIAL_HDWALLET_STATE)
+
+const connectBTC = async () => {
```

Comment:
at our business logic layer we don't use promises but use streams. could you use it here